runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java

```diff
@@ -318,6 +328,33 @@ public void onItemContainerChanged(final ItemContainerChanged event)
 
 		inventoryItems = event.getItemContainer().getItems();
 
+		if (Arrays.stream(inventoryItems).anyMatch(i -> i.getId() == ItemID.RUNE_POUCH || i.getId() == ItemID.RUNE_POUCH_L))
+		{
+			List<Item> runePouchContents = getRunepouchContents();
+
+			for (int i = 0; i < inventoryItems.length; i++)
+			{
+				Item invItem = inventoryItems[i];
+				for (Item rune : runePouchContents)
+				{
+					if (invItem.getId() == rune.getId())
+					{
+						inventoryItems[i] = new Item(invItem.getId(), rune.getQuantity() + invItem.getQuantity());
+						runePouchContents.remove(rune);
+						break;
+					}
+				}
+				if (runePouchContents.size() <= 0)
```

Comment:
Instead of being a `break` check, this can be incorporated into the loop condition:

```
for (int i = 0; i < inventoryItems.length && runePouchContents.size() > 0; i++)
```